docs/data/base/components/switch/switch.md

```diff
@@ -22,23 +21,40 @@ To make the switch accessible, you should ensure the corresponding label reflect
 import SwitchUnstyled from '@mui/base/SwitchUnstyled';
 ```
 
-The `SwitchUnstyled` component provides default components and assigns CSS classes you can style entirely on your own.
-You are free to choose any styling solution - plain CSS classes, a CSS framework, Emotion, etc.
-It is also possible to replace these default components by other HTML elements or custom components.
+## Basic switch
 
-There are three components you can override by the `components` prop: `Root`, `Thumb` and `Input`. Each one's props can be set using the `componentsProps` object.
+The `SwitchUnstyled` component is composed of a root `<span>` that houses three interior elements—a track, a thumb, and an input—and it assigns CSS classes for styling each piece:
+
+```html
+<span class="MuiSwitch-root Mui-checked">
+  <span class="MuiSwitch-track"></span>
+  <span class="MuiSwitch-thumb"></span>
+  <input type="checkbox" aria-label="Demo switch" class="MuiSwitch-input" checked />
+</span>
+```
+
+You can set the props for these interior components using the `componentsProps` object.
+
+You can also override them entirely with the `components` prop, to replace them with other HTML elements or custom components.
+
+The following demo shows how to assign styles and props to the interior elements of the `SwitchUnstyled` component:
 
 {{"demo": "UnstyledSwitches.js"}}
 
-## useSwitch
+## Accessibility
 
-For the ultimate customizability, a `useSwitch` hook is available.
-It accepts almost the same options as the SwitchUnstyled component minus the `component`, `components`, and `componentsProps` props.
+To make the `SwitchUnstyled` component accessible, you should ensure that the corresponding labels reflect the current state of the switch.
 
-```jsx
+## The useSwitch hook
```

Comment:
I wonder if we shouldn't wrap hooks (or anything in camel case) in code tags. What do you think about that?

Reply:
I would also suggest to keep it generic so it scales to components that are built out of 3,4,5 hooks

```suggestion
## Hook API 
```

Reply:
I think having the name in the header is preferable, but I don't know how much of a difference it makes.

If we go with what @oliviertassinari proposes, then I'm thinking we should do something like this:

## Hooks
### useSwitch

...this is in line with the bigger picture formatting changes we were discussing last week, which would look something like this for the pages as a whole:

 ## Introduction
 ## Components API
 ### Component only content
 ## Hooks
 ### Hooks only content
 
In any case, I think that since this is a big-picture question that impacts all of these docs, we should leave this one alone for now and address the formatting in a separate PR that can cover all the pages at once.